tests/io/FileFactoryTest.py

```diff
@@ -0,0 +1,46 @@
+import os
+import unittest
+
+from coalib.io.FileFactory import FileFactory
+
+
+class FileFactoryTest(unittest.TestCase):
+
+    def setUp(self):
+        file_factory_test_dir = os.path.join(os.path.split(__file__)[0],
+                                             'FileFactoryTestFiles')
+
+        self.test_file = os.path.join(file_factory_test_dir, 'test.txt')
+        self.uut = FileFactory(self.test_file)
+
+    def test_equal(self):
+        self.assertEqual(self.uut, FileFactory(self.test_file))
+
+    def test_iter(self):
+        self.uut.lines
+        for f in self.uut:
+            self.assertEqual(type(f), str)
```

Comment:
I would rather do
```python
self.assertEqual(list(self.uut), ['This is a test file.'])
```
Easier and better test, because you assert for the result ;)